### DISCUSSION TITLE
Frigate Object Mask Help Request

### DISCUSSION THREAD

**Discussion by cypher2001** [blakeblackshear/frigate#6760] [Ask A Question]
Hey Folks-

  I'm looking for some help on how to setup my mask to avoid Frigate from detecting a bush as a person.   i've setup all kinds of masks and none seem to be working to avoid detecting that area.   I've also tried in vain with zones.  Looking for any help you can offer to get this to stop showing me hundreds of errant records a day.

 `  GarageCam2:
    ffmpeg:
      inputs:
        - path: ******
          roles:
            - detect
            - record
    zones:
      FrontGarden:
        coordinates: 1873,1305,2017,1711,2264,1726,2633,1788,3119,1954,3437,2139,3820,2124,3820,1549,3614,1316,3094,1073,2777,1014,1951,914
        #objects: 
      

    detect:
      enabled: true
      width: 3820
      height: 2160

    objects:
        filters:
          person:
           mask:
            - 3481,2160,3791,2076,3794,1560,3632,1324,3057,1091,2411,992,2175,918,1914,1025,1855,1294,2013,1733
            - 1504,1822,1641,1833,1862,1818,2013,1799,2356,1788,2323,1667,1530,1604
            - 1869,900,2113,878,2434,937,2600,1202,2467,1895,1501,1877
objects:
 track:
   - person
   - bicycle
   - fox
   - cat
   - dog
   `

Currently I have multiple masks enabled but this should have covered it all.

![image](https://github.com/blakeblackshear/frigate/assets/22996629/86e1c3fb-1016-4b33-9b06-9fcd704545f9)

Let me know what other details might be helpful here.  Admittedly I'm somewhat of a newbie so its very possible its a simple fix. (Hopeful)

Thanks in advance

**Comment by NickM-27** [comment]
Like it says in the docs, object masks block objects based on the bottom center pixel of their bounding box. You'd likely just need a single mask at the bottom edge of the bush.

**Comment by cypher2001** [comment]
Thank you @NickM-27 
Appreciate the response.  So you are suggesting a mask similar to this?

![image](https://github.com/blakeblackshear/frigate/assets/22996629/97421422-4708-4386-9b63-b9f9071f600c)

**Reply by NickM-27** [reply]
yes, and then based on if it still gets detected look at the bottom center of the bounding box and see if the mask needs to be expanded in some way

**Comment by cypher2001** [comment]
Based on that update, I'm still getting detections:

![image](https://github.com/blakeblackshear/frigate/assets/22996629/b2ee120b-74f8-47aa-8454-28d644b077e2)

Suggest I move the box down lower?  Trying to figure out where the bottom of the selection is to adjust the box

**Reply by NickM-27** [reply]
You should be using the debug live view or enable bounding boxes on snapshots.

**Comment by cypher2001** [comment]
I've been watching the debug view but there are boxes all over the place.  Its hard to distinguish which one is triggering the person.  As for the bounding boxes on snapshots, I'm not seeing  a way to enable.
Again, sorry for the basic questions..  

I'm a little confused if I have a large mask why it wouldnt care for those triggers.  Unless the bounding box is lower than my original mask.   I feel like I want to add a large box and slowly make it smaller until its as selective as I can be.

**Reply by NickM-27** [reply]
> I'm a little confused if I have a large mask why it wouldnt care for those triggers. Unless the bounding box is lower than my original mask. I feel like I want to add a large box and slowly make it smaller until its as selective as I can be.

but this is just guessing. There is no point in doing that when you can specifically check and see the spot that triggered it. 

> Its hard to distinguish which one is triggering the person. As for the bounding boxes on snapshots, I'm not seeing a way to enable.
Again, sorry for the basic questions..

```
snapshots:
  enabled: true
  bounding_box: true
```

**Reply by cypher2001** [reply]
Thank you.  Seems like I'm on the path to getting this cleared up.  Appreciate the help!

Not sure why the object recognition is giving so many false positives for people.    Its only on that camera where I have so many.  Others only have the occasional misque.
